### ISSUE TITLE
deactivate env fails on OS X for PyStan (uses new compilers)

### ISSUE THREAD

**Issue by tloredo** [conda/conda#6617]
I'm attempting to create an environment in which to use PyStan. PyStan creates Python C/C++ modules on-the-fly and so uses the Anaconda compilers. I'm trying to keep my base env clean in terms of paths to standard compilers (like clang); the PyStan env changes `clang` to point to Anaconda's compiler rather than Apple's.

I have Miniconda (Py-3) newly installed, and I created the env this way:

`conda create -n py36stan python=3.6 anaconda pystan`

Activating it appears to work fine:

```
$ source activate py36stan
INFO: activate_clang_osx-64.sh made the following environmental changes:
+AR=/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-ar
+AS=/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-as
+CC=x86_64-apple-darwin13.4.0-clang
+CFLAGS=-march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe
+CHECKSYMS=/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-checksyms
+CLANG=/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-clang
+CODESIGN_ALLOCATE=/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-codesign_allocate
+CONDA_BUILD_SYSROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk
...
```

Deactivating fails:

```
$ source deactivate
ERROR: This cross-compiler package contains no program /bin/x86_64-apple-darwin13.4.0-ar
ERROR: deactivate_clang_osx-64.sh failed, see above for details
ERROR: This cross-compiler package contains no program /bin/x86_64-apple-darwin13.4.0-clang++
ERROR: deactivate_clangxx_osx-64.sh failed, see above for details

$ echo $AR
/Users/tester/miniconda3/envs/py36stan/bin/x86_64-apple-darwin13.4.0-a
```

Note the last; $AR did not exist in the base env; it is created for the new one and removing it is evidently aborted when `deactivate` fails.

I find the error messages unclear. I don't know what package they refer to. I also don't know if they refer to a problem in the new py36stan env or in the base env.

I realize Anaconda 5 has new, arch-specific compilers, implementing a kind of cross-compilation capability:  [Utilizing the new compilers...](https://www.anaconda.com/blog/developer-blog/utilizing-the-new-compilers-in-anaconda-distribution-5/?lang=en-us). I don't fully understand when they need to be used and when/whether the Apple compilers mentioned on that page need to be installed. I do know that `clang_osx-64` and `clang_osx-64` are installed as PyStan dependencies, and that they put a bunch of `x86_64-apple-darwin...` tools in the /bin directory of the env. I tried installing these two clang packages in my base env before creating the PyStan env, but the deactivate failure still happens.

Is there something I'm neglecting to do with `conda`, or is this is a bug?

**Comment by mingwandroid** [comment]
It is a bug in conda that's been fixed in 4.4.5 which is available on the conda canary channel. I hope a new proper release comes soon as this bug is a blocker. Ping @kalefranz.

You shouldn't use Apple's compiler tools with the Anaconda Distribution. Ours are better (they support openmp for example and clang 4.0.1 on macOS 10.9) and you avoid any potential incompatibility be sticking with ours instead of mixing compilers.

**Comment by kalefranz** [comment]
4.4.5 won’t be released to defaults. 4.4.6 is only gated by one PR pending review labeled with !blocking!

**Comment by mingwandroid** [comment]
@tloredo, in the meantime, to get you unstuck you can do the following (from an activated root env Terminal):

```
pushd $CONDA_PREFIX/lib/python-3.6/site-packages/conda
curl https://github.com/conda/conda/commit/1cf2b2b60fcb302c47e8167efef6edf87c4e0d45.patch | patch -tp2
```

It should report:

```
patching file activate.py
can't find file to patch at input line 39
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/tests/test_activate.py b/tests/test_activate.py
|index 12535eff1..cc1a8c797 100644
|--- a/tests/test_activate.py
|+++ b/tests/test_activate.py
--------------------------
No file to patch.  Skipping patch.
14 out of 14 hunks ignored
```

Don't worry about the failure to patch `test_activate.py`, so long as `activate.py` gets patched you should be ok.

**Comment by tloredo** [comment]
@mingwandroid, @kalefranz: Thank you for the quick and helpful responses, esp. in light of the holiday timing!

> You shouldn't use Apple's compiler tools with the Anaconda Distribution.

I confess I'm confused about the compiler situation under Anaconda 5. I suppose this isn't a bug question and so the following remarks perhaps belong on the conda users group. I did search there for recent posts about the Anaconda compilers, but there is nothing on compilers since 2014. I'll ask some questions here, but if this should be redirected elsewhere, let me know where. It's probably a documentation issue, not a conda issue per se; I don't know if conda documentation issues belong in this repo or elsewhere.

Although in the past I've used Anaconda as my root environment, I'm now setting up software for a workshop and seeking a setup that doesn't impact users' non-conda development. So I'm starting with miniconda specifically to leave command line compilers alone in the base environment. But we'll always be working with Anaconda's Python in other environments, so I want those to be set up properly.

As a first quick question, the initial environment I have with a new miniconda is called "base", not "root" (as in my other Anaconda-based installations). Is there something significant about the name change? The conda docs say that the initial environment is called "root". It may be worth mentioning this isn't the case for miniconda in the docs, esp. since the docs appear to cover both miniconda and anaconda: [Conda docs](https://conda.readthedocs.io/en/latest/#getting-started).

After installing miniconda, I created a basic "py36" environment by installing just anaconda (with Python specified as 3.6). In that environment, I find that the Anaconda compilers are *not* installed. That environment does include `cython`, so I presume it's ready for creating C extensions that are compatible with Anaconda's Python and PyData stack. I do develop and use C extensions, so I do need an environment supporting that. This leads to two questions:

1. Does the Anaconda distribution by itself support building/creating C extensions? I presume these would be using Apple's compilers.

2. Even if the answer to (1) is "yes," should I be installing the Anaconda compilers anyway (for performance/security reasons)? The [new compilers blog post](https://www.anaconda.com/blog/developer-blog/utilizing-the-new-compilers-in-anaconda-distribution-5/?lang=en-us) suggests as much. If this is the case, it seems to me this information really should be given prominently in the docs, and not just in a blog post.

If the answer to (2) is "yes", then I presume *best practice* (but not a requirement) for Mac users is to always install Anaconda with clang_osx-64, clangxx_osx-64, and gfortran_osx-64 (I do also develop Fortran extensions). Is this right? If so, it seems to me that this should be stated prominently somewhere in the conda docs (and maybe even on the Anaconda installation page, since users not using environments will probably assume their installation is "batteries included").

Finally, the most confusing thing about the new compilers blog post is the sections about installing MacOS SDK and backward compatibility. If I install Anaconda's compilers with my Anaconda installations, is it the case that I don't have to worry about these two sections (i.e., they are addressed only to those *not* using the new Anaconda compilers)? Alternatively, if I install only the `anaconda` package, and I have the latest Xcode and its command-line tools installed, is there something else I need to do in order to support extensions compatible with Anaconda's Python? The "backward compatibility" section suggests not, but the "MacOS SDK" section suggests I need to install a 10.9-compatible SDK.

I know I've asked a lot here; even partial responses would be appreciated!

**Comment by mingwandroid** [comment]
It's approaching midnight  (~4 hours to go) where I am so I'll not reply properly until tomorrow if that's ok? But in general yes, use our compilers. That's more important for people sharing built binaries but it's just safer in general. As for the SDK, use a 10.9 one if you can and set CONDA_BUILD_SYSROOT to the location of that SDK.

**Comment by tloredo** [comment]
@mingwandroid: Ray, I wasn't expecting any response until tomorrow! Thanks for checking in anyway and happy new year. I'd appreciate an additional response eventually, but if you think this discussion belongs elsewhere, let me know (tomorrow) and I'll move it. As for the original bug, I'm satisfied at this point and the issue can be closed if you see fit.

**Comment by tloredo** [comment]
@kalefranz: Regarding this:

>4.4.5 won’t be released to defaults. 4.4.6 is only gated by one PR...

I just tried a `conda update conda` on a machine with Anaconda-5.0.1 in the root env:

```
The following packages will be UPDATED:

    anaconda: 5.0.1-py27he797f56_1 --> custom-py27h2cfa9e9_0
    conda:    4.3.31-py27_0        --> 4.4.6-py27_0
    pycosat:  0.6.2-py27h085d4cc_0 --> 0.6.3-py27h6c51c7e_0
...
```

I declined the update because I'm doubly confused:

* I see it will make my anaconda distribution "custom." I don't know how that might affect subsequent updates of anaconda. If my version of it is "custom," will future updates get me back in sync with future Anaconda versions? Or will conda consider me to have defined my own "custom" distribution and thus start ignoring anaconda-5.x specifications?

* I checked [conda releases on GitHub](https://github.com/conda/conda/releases) and it indicates the current release is 4.4.5, not 4.4.6. Further, the [conda 4.4.6 milestones page](https://github.com/conda/conda/milestone/13) indicates the milestones for 4.4.6 are still incomplete. How is this consistent with `conda update` offering to update me to to 4.4.6? That you mentioned that 4.4.6 was still blocked just 18 h ago made me further suspicious about this apparent inconsistency.

Clarification would be appreciated.

**Comment by kalefranz** [comment]
You can probably tease out the answers to your questions about Anaconda-custom in https://github.com/ContinuumIO/anaconda-issues/issues/4298

Regarding the update, I just pushed it. Not everything happens at the same time. I guess if there’s a single source of truth, it’s whats actually in the repo.

**Comment by kalefranz** [comment]
All indicators should now be aligned for 4.4.6.

**Comment by kalefranz** [comment]
Regarding specifics on the switch of anaconda to the version `custom`, all packages listed within the `anaconda` metapackage have versions that are explicitly pinned.  I guess we include `pyocosat` in the anaconda metapackage, even though we probably shouldn't, because I'm pretty sure conda is the ONLY package that needs it, and conda isn't in the anaconda metapackage.

Anyway, digression.  Since pycosat is pinned to 0.6.2 in the latest anaconda metapackage, and conda 4.4 needs pycosat 0.6.3, then the version of the anaconda metapackage flips to custom.  It's a total hack, but something that's gotten us out of a bind for the last 18 months or so.  Still, there are deeper issues that were the root cause of why the hack was even needed to begin with, and that's what ContinuumIO/anaconda-issues#4298 is meant to address in the future.

In short, the switch to custom is expected, and conda is working correctly.

**Comment by tloredo** [comment]
Thanks for the clarifications. I presumed the switch to custom was expected; I just wanted to make sure that it would switch back to "regular" anaconda some time in the future (when the pycosat versions are aligned again). In other words, I'm assuming conda isn't really attaching a "custom" label to my installation, but rather *recognizing* it as customized, and able to recognize it as "regular" in the future when it becomes aligned.

**Comment by kalefranz** [comment]
Actually, it's pulling in an `anaconda` package with version `custom`, which ranks at basically version 0, or lowest in the sort order.  So the solver will reach all the way back to that if it absolutely *has* to in order to satisfy what you've given it.  See below, from https://repo.continuum.io/pkgs/main/osx-64/repodata.json

The `anaconda-custom` package works based on the assumption that you already have "all the anaconda you need" installed in the environment.  

```
    "anaconda-custom-py27h2cfa9e9_0.tar.bz2": {
      "build": "py27h2cfa9e9_0",
      "build_number": 0,
      "depends": [
        "python >=2.7,<2.8.0a0"
      ],
      "license": "BSD",
      "md5": "0f30a84f4bcf580892f685e6881cc6f6",
      "name": "anaconda",
      "sha256": "e5ceb73f267b8f5813dec8e43846eb1da8d508599ff1d7b8f6d5ea1bd2dca91d",
      "size": 5657,
      "subdir": "osx-64",
      "timestamp": 1508808529637,
      "version": "custom"
    },
```

```
    "anaconda-5.0.1-py27he797f56_1.tar.bz2": {
      "build": "py27he797f56_1",
      "build_number": 1,
      "depends": [
        "alabaster 0.7.10 py27h9dd7d6e_0",
        "anaconda-client 1.6.5 py27hc13fba8_0",
        "anaconda-navigator 1.6.9 py27h103b016_0",
        "anaconda-project 0.8.0 py27h9e3d455_0",
        "appnope 0.1.0 py27hb466136_0",
        "appscript 1.0.1 py27h451298e_1",
        "asn1crypto 0.22.0 py27h61af4a7_1",
        "astroid 1.5.3 py27h96f3fd4_0",
        "astropy 2.0.2 py27h87cc2bd_4",
        "babel 2.5.0 py27h7311c9e_0",
        "backports 1.0 py27hb4f9756_1",
        "backports.functools_lru_cache 1.4 py27h2aca819_1",
        "backports.shutil_get_terminal_size 1.0.0 py27hc9115de_2",
        "backports_abc 0.5 py27h6972548_0",
        "beautifulsoup4 4.6.0 py27h9416283_1",
        "bitarray 0.8.1 py27hd5bfd95_0",
        "bkcharts 0.2 py27haafc882_0",
        "blaze 0.11.3 py27hb49378a_0",
        "bleach 2.0.0 py27ha7d1710_0",
        "bokeh 0.12.10 py27hf91827b_0",
        "boto 2.48.0 py27hacdd0fd_1",
        "bottleneck 1.2.1 py27h71f98a3_0",
        "bzip2 1.0.6 h92991f9_1",
        "ca-certificates 2017.08.26 ha1e5d58_0",
        "cdecimal 2.3 py27hf5d9fd9_1",
        "certifi 2017.7.27.1 py27h482ffc0_0",
        "cffi 1.10.0 py27haac214c_1",
        "chardet 3.0.4 py27h2842e91_1",
        "click 6.7 py27h2b86a94_0",
        "cloudpickle 0.4.0 py27h665dddb_0",
        "clyent 1.2.2 py27hc0ae608_0",
        "colorama 0.3.9 py27hbbe92b6_0",
        "configparser 3.5.0 py27hc7edf1b_0",
        "contextlib2 0.5.5 py27h9cb85f4_0",
        "cryptography 2.0.3 py27hab69567_1",
        "curl 7.55.1 h7601780_3",
        "cycler 0.10.0 py27hfc73c78_0",
        "cython 0.26.1 py27h6a053f9_0",
        "cytoolz 0.8.2 py27ha56cda1_0",
        "dask 0.15.3 py27h8a1b457_0",
        "dask-core 0.15.3 py27h6a32bf4_0",
        "datashape 0.5.4 py27hd6a1745_0",
        "dbus 1.10.22 h50d9ad6_0",
        "decorator 4.1.2 py27h9f877ea_0",
        "distributed 1.19.1 py27h67dd2ec_0",
        "docutils 0.14 py27h0befae3_0",
        "entrypoints 0.2.3 py27hd680fb1_2",
        "enum34 1.1.6 py27hf475452_1",
        "et_xmlfile 1.0.1 py27hc42f929_0",
        "expat 2.2.4 h8f26bf8_1",
        "fastcache 1.0.2 py27hc4635c7_0",
        "filelock 2.0.12 py27h72fe922_0",
        "flask 0.12.2 py27h3ac5568_0",
        "flask-cors 3.0.3 py27h13db576_0",
        "freetype 2.8 h143eb01_0",
        "funcsigs 1.0.2 py27hb9f6266_0",
        "functools32 3.2.3.2 py27h8ceab06_1",
        "futures 3.1.1 py27hb02a37a_0",
        "get_terminal_size 1.0.0 h7520d66_0",
        "gettext 0.19.8.1 hb0f4f8b_2",
        "gevent 1.2.2 py27hc02608c_0",
        "glib 2.53.6 ha08cb78_1",
        "glob2 0.5 py27h2f8fe13_0",
        "gmp 6.1.2 h4a9834d_0",
        "gmpy2 2.0.8 py27h7e2fca4_1",
        "greenlet 0.4.12 py27h081ed54_0",
        "grin 1.2.1 py27hc43e5f3_1",
        "h5py 2.7.0 py27h217cc45_1",
        "hdf5 1.10.1 h6090a45_0",
        "heapdict 1.0.0 py27hb5e74ad_0",
        "html5lib 0.999999999 py27hec7e2bc_0",
        "icu 58.2 hea21ae5_0",
        "idna 2.6 py27hedea723_1",
        "imageio 2.2.0 py27h37746d9_0",
        "imagesize 0.7.1 py27h4f7bcc8_0",
        "intel-openmp 2018.0.0 h68bdfb3_7",
        "ipaddress 1.0.18 py27h5b9a5b9_0",
        "ipykernel 4.6.1 py27h1e70a78_0",
        "ipython 5.4.1 py27h2b3d779_1",
        "ipython_genutils 0.2.0 py27h8b9a179_0",
        "ipywidgets 7.0.0 py27h3e52029_0",
        "isort 4.2.15 py27h5bf637f_0",
        "itsdangerous 0.24 py27h3948ded_1",
        "jbig 2.1 h4d881f8_0",
        "jdcal 1.3 py27hfeaf94f_0",
        "jedi 0.10.2 py27hbb5dc62_0",
        "jinja2 2.9.6 py27h92590e2_1",
        "jpeg 9b haccd157_1",
        "jsonschema 2.6.0 py27hd9b497e_0",
        "jupyter 1.0.0 py27hec63c99_0",
        "jupyter_client 5.1.0 py27hfaf569a_0",
        "jupyter_console 5.2.0 py27h9702a86_1",
        "jupyter_core 4.3.0 py27hd5161ba_0",
        "jupyterlab 0.27.0 py27h25d4955_2",
        "jupyterlab_launcher 0.4.0 py27he518b91_0",
        "lazy-object-proxy 1.3.1 py27h712ce3f_0",
        "libcxx 4.0.1 h579ed51_0",
        "libcxxabi 4.0.1 hebd6815_0",
        "libedit 3.1 hb4e282d_0",
        "libffi 3.2.1 hd939716_3",
        "libgfortran 3.0.1 h93005f0_2",
        "libiconv 1.15 h99df5da_5",
        "libpng 1.6.32 hce72d48_2",
        "libsodium 1.0.13 hba5e272_2",
        "libssh2 1.8.0 h1218725_2",
        "libtiff 4.0.8 h8cd0352_9",
        "libxml2 2.9.4 hbd0960b_5",
        "libxslt 1.1.29 h95a2935_5",
        "llvmlite 0.20.0 py27_0",
        "locket 0.2.0 py27ha10513d_1",
        "lxml 4.1.0 py27hcb5f3a6_0",
        "lzo 2.10 hb6b8854_1",
        "markupsafe 1.0 py27hd3c86fa_1",
        "matplotlib 2.1.0 py27h3078195_0",
        "mccabe 0.6.1 py27h1f69e8d_0",
        "mistune 0.7.4 py27h1658d75_0",
        "mkl 2018.0.0 h5ef208c_6",
        "mkl-service 1.1.2 py27hcba3029_4",
        "mpc 1.0.3 hc455b36_4",
        "mpfr 3.1.5 h7fa3772_1",
        "mpmath 0.19 py27h09cdc99_2",
        "msgpack-python 0.4.8 py27h635ded4_0",
        "multipledispatch 0.4.9 py27h10993aa_0",
        "navigator-updater 0.1.0 py27ha63e0b4_0",
        "nbconvert 5.3.1 py27h6455e4c_0",
        "nbformat 4.4.0 py27hddc86d0_0",
        "ncurses 6.0 ha932d30_1",
        "networkx 2.0 py27h2503496_0",
        "nltk 3.2.4 py27h1626047_0",
        "nose 1.3.7 py27h2ee3cb8_2",
        "notebook 5.0.0 py27h5f5981d_2",
        "numba 0.35.0 np113py27_6",
        "numexpr 2.6.2 py27h99e4b4f_1",
        "numpy 1.13.3 py27h62f9060_0",
        "numpydoc 0.7.0 py27h022f19e_0",
        "odo 0.5.1 py27h992a9f7_0",
        "olefile 0.44 py27h73ba740_0",
        "openpyxl 2.4.8 py27h70e7ed9_1",
        "openssl 1.0.2l h57f3a61_2",
        "packaging 16.8 py27h24b219a_0",
        "pandas 0.20.3 py27h2a3d8ca_2",
        "pandoc 1.19.2.1 ha5e8f32_1",
        "pandocfilters 1.4.2 py27hed78c4e_1",
        "partd 0.3.8 py27h7560dbf_0",
        "path.py 10.3.1 py27h5e25276_0",
        "pathlib2 2.3.0 py27he09da1e_0",
        "patsy 0.4.1 py27h40ed276_0",
        "pcre 8.41 h29eefc5_0",
        "pep8 1.7.0 py27hff3397c_0",
        "pexpect 4.2.1 py27hc4e4961_0",
        "pickleshare 0.7.4 py27h37e3d41_0",
        "pillow 4.2.1 py27h2cf1d5f_0",
        "pip 9.0.1 py27h61def0c_3",
        "pkginfo 1.4.1 py27ha9221e7_0",
        "ply 3.10 py27h6279b8a_0",
        "prompt_toolkit 1.0.15 py27h4a7b9c2_0",
        "psutil 5.4.0 py27h7076ac6_0",
        "ptyprocess 0.5.2 py27h70f6364_0",
        "py 1.4.34 py27h917aa57_1",
        "pyaudio 0.2.7 py27h3777516_1",
        "pycodestyle 2.3.1 py27h5b634e0_0",
        "pycosat 0.6.2 py27h085d4cc_0",
        "pycparser 2.18 py27h0d28d88_1",
        "pycrypto 2.6.1 py27h4efa152_1",
        "pycurl 7.43.0 py27h398a7fe_3",
        "pyflakes 1.6.0 py27h4446e76_0",
        "pygments 2.2.0 py27h1a556bb_0",
        "pylint 1.7.4 py27hc678664_0",
        "pyodbc 4.0.17 py27hc9de18c_0",
        "pyopenssl 17.2.0 py27h732fe57_0",
        "pyparsing 2.2.0 py27h5bb6aaf_0",
        "pyqt 5.6.0 py27hf21fe59_6",
        "pysocks 1.6.7 py27h1cff6a6_1",
        "pytables 3.4.2 py27ha4551b8_2",
        "pytest 3.2.1 py27h173df03_1",
        "python 2.7.14 he768d2d_19",
        "python-dateutil 2.6.1 py27hd56c96b_1",
        "python.app 2 py27h84f57d0_6",
        "pytz 2017.2 py27hb891d23_1",
        "pywavelets 0.5.2 py27hd99e88a_0",
        "pyyaml 3.12 py27ha7932d0_1",
        "pyzmq 16.0.2 py27he61c07e_2",
        "qt 5.6.2 h9975529_14",
        "qtawesome 0.4.4 py27hdeb2f59_0",
        "qtconsole 4.3.1 py27hdc90b4f_0",
        "qtpy 1.3.1 py27h39159f8_0",
        "readline 7.0 h81b24a6_3",
        "requests 2.18.4 py27h9b2b37c_1",
        "rope 0.10.5 py27he855d02_0",
        "ruamel_yaml 0.11.14 py27h31666c4_2",
        "scandir 1.6 py27h97aa1ee_0",
        "scikit-image 0.13.0 py27h03e84e1_1",
        "scikit-learn 0.19.1 py27h9788993_0",
        "scipy 0.19.1 py27hf01dd8f_3",
        "seaborn 0.8.0 py27h92884e4_0",
        "setuptools 36.5.0 py27h2a45cec_0",
        "simplegeneric 0.8.1 py27h6db5e31_0",
        "singledispatch 3.4.0.3 py27he22c18d_0",
        "sip 4.18.1 py27h6300f65_2",
        "six 1.11.0 py27h7252ba3_1",
        "snowballstemmer 1.2.1 py27h68ac032_0",
        "sortedcollections 0.5.3 py27h8094be4_0",
        "sortedcontainers 1.5.7 py27h322dbbf_0",
        "sphinx 1.6.3 py27h11269f0_0",
        "sphinxcontrib 1.0 py27hd2ed746_1",
        "sphinxcontrib-websupport 1.0.1 py27h857890b_1",
        "spyder 3.2.4 py27h93d9c3e_0",
        "sqlalchemy 1.1.13 py27hcbc9ed3_0",
        "sqlite 3.20.1 h900c3b0_1",
        "ssl_match_hostname 3.5.0.1 py27h8780752_2",
        "statsmodels 0.8.0 py27h6d68dbf_0",
        "subprocess32 3.2.7 py27h24b2887_0",
        "sympy 1.1.1 py27hce55102_0",
        "tblib 1.3.2 py27ha684fc4_0",
        "terminado 0.6 py27he40bf16_0",
        "testpath 0.3.1 py27h72d81a5_0",
        "tk 8.6.7 hcdce994_1",
        "toolz 0.8.2 py27h27228c4_0",
        "tornado 4.5.2 py27h29aec9e_0",
        "traitlets 4.3.2 py27hcf08151_0",
        "typing 3.6.2 py27h646fea0_0",
        "unicodecsv 0.14.1 py27h170f95c_0",
        "unixodbc 2.3.4 h4cb4dde_1",
        "urllib3 1.22 py27hc3787e9_0",
        "wcwidth 0.1.7 py27h817c265_0",
        "webencodings 0.5.1 py27h19a9f58_1",
        "werkzeug 0.12.2 py27hcac71f8_0",
        "wheel 0.29.0 py27h84bd1c0_1",
        "widgetsnbextension 3.0.2 py27h56f70de_1",
        "wrapt 1.10.11 py27hd341262_0",
        "xlrd 1.1.0 py27hbd41ed1_1",
        "xlsxwriter 1.0.2 py27h7f1064a_0",
        "xlwings 0.11.4 py27h4d78f01_0",
        "xlwt 1.2.0 py27hbeec4ae_0",
        "xz 5.2.3 ha24016e_1",
        "yaml 0.1.7 hff548bb_1",
        "zeromq 4.2.2 h131e0f7_1",
        "zict 0.1.3 py27h5fff8b1_0",
        "zlib 1.2.11 h60db283_1"
      ],
      "license": "BSD",
      "md5": "ba95c8da2cac5b1f5b05a70525efadb6",
      "name": "anaconda",
      "sha256": "b6876e3da66690c549113ced4064e75bdddfc31ee29d43e049e2b5ef89c762a5",
      "size": 13466,
      "subdir": "osx-64",
      "timestamp": 1508801419626,
      "version": "5.0.1"
    },
```

**Comment by tloredo** [comment]
Thanks for that; for someone who has never looked into how large packages like this are maintained, this is revealing.

**Comment by kalefranz** [comment]
As I'm reading it, this issues seems resolved and can be closed.

**Comment by braaannigan** [comment]
Hi @kalefranz 
I'm getting a similar issue with conda version 4.4.7.  I create an environment that includes pystan.  When I activate the environment I get a lot of messages related to setting up the compiler including:
+_PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_x86_64_apple_darwin13_4_0

.  However, when I try to run python/ipython/jupyter I get the following error (I'm not sure why markdown is creating strikethrough lines across the text below):
'''
Failed to import the site module
Traceback (most recent call last):
  File "~/anaconda3/envs/bayes/lib/python3.6/site.py", line 544, in <module>
    main()
  File "/~/anaconda3/envs/bayes/lib/python3.6/site.py", line 530, in main
    known_paths = addusersitepackages(known_paths)
  File "/~/anaconda3/envs/bayes/lib/python3.6/site.py", line 282, in addusersitepackages
    user_site = getusersitepackages()
  File "/~/anaconda3/envs/bayes/lib/python3.6/site.py", line 258, in getusersitepackages
    user_base = getuserbase() # this will also set USER_BASE
  File "/~/anaconda3/envs/bayes/lib/python3.6/site.py", line 248, in getuserbase
    USER_BASE = get_config_var('userbase')
  File "/~/anaconda3/envs/bayes/lib/python3.6/sysconfig.py", line 601, in get_config_var
    return get_config_vars().get(name)
  File "/~/anaconda3/envs/bayes/lib/python3.6/sysconfig.py", line 550, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/~/anaconda3/envs/bayes/lib/python3.6/sysconfig.py", line 421, in _init_posix
    _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
ModuleNotFoundError: No module named '_sysconfigdata_x86_64_apple_darwin13_4_0'
'''

**Comment by mingwandroid** [comment]
You need to update the Python in your root environment.
